docs/content/concepts/sui-move-concepts/packages.mdx

```diff
@@ -1,6 +1,34 @@
 ---
-title: Packages
-draft: true
+title: Package Upgrades
+description: A Move package on Sui includes one or more modules that define that package's interaction with on-chain objects. Upgrading on-chain packages provides a way to improve your code or add features without affecting packages that use your published modules.
 ---
 
-Content coming soon
\ No newline at end of file
+A Move package on Sui includes one or more modules that define that package's interaction with on-chain objects. You develop the logic for those modules using Move, which you then compile into an object. Finally, you publish your package object onto a Sui network. On chain, anyone can view your package contents and the logic it employs to manipulate other on-chain objects using a Sui network explorer.
+
+## Lifecycle of a package
+
+After you publish a package object on chain to Mainnet, it lives there forever. You cannot directly change the code of an on-chain package object. After a package object appears on chain, other packages can use the modules that the original package provides. Bad actors aside, imagine if someone changed the logic in their on-chain package to correct unintentional behavior that was overlooked during development. This would cause a ripple effect, changing the logic of every package that uses that module to perform in ways the developers might have never intended.
```

Comment:
It is, but that's a kind of external factor -- in some sense, mainnet doesn't get reset because validators choose not to reset it.  I.e. to my mind there's a distinction between what the protocol controls (in which case package objects are immutable) and things outside the protocol (like the devnet and testnet reset schedules).